modules/simple_forms_api/app/services/simple_forms_api/benefits_intake_client.rb

```diff
@@ -0,0 +1,192 @@
+    end
+
+    private
+
+    attr_accessor :attachments, :file_path, :form_id, :form, :metadata, :params
+
+    def lighthouse_service
+      @lighthouse_service ||= BenefitsIntake::Service.new
+    end
+
+    def fetch_form_id
+      FORM_NUMBER_MAP.fetch(form_number) { raise ArgumentError, "Invalid form_number: #{form_number}" }
+    end
+
+    def parse_and_assign_form_data
+      JSON.parse(params.to_json).tap do |parsed_form_data|
+        assign_form_info(parsed_form_data)
+      end
+    end
+
+    def log_submission(status, uuid)
+      Rails.logger.info('PDF was successfully uploaded to benefits intake', { form_number: form_id, status:, uuid: })
+    end
+
+    def send_confirmation_email(parsed_form_data, confirmation_number, status)
+      return unless status == 200 && Flipper.enabled?(:simple_forms_email_confirmations)
+
+      SimpleFormsApi::ConfirmationEmail.new(
+        form_data: parsed_form_data,
+        form_number: form_id,
+        confirmation_number:,
+        user: @current_user
+      ).send
+    end
+
+    def generate_response(confirmation_number, status)
+      { json: confirmation_payload(confirmation_number), status: }
+    end
+
+    def confirmation_payload(confirmation_number)
+      { confirmation_number: }.tap do |payload|
+        payload[:expiration_date] = 1.year.from_now if form_id == 'vba_21_0966'
+      end
+    end
+
+    def assign_form_info(parsed_form_data)
+      @form = initialize_form(parsed_form_data)
+      @file_path = generate_filled_form
+      @metadata = validate_metadata
+
+      handle_form_specific_logic
+    end
+
+    def initialize_form(parsed_form_data)
+      form_class = "SimpleFormsApi::#{form_id.titleize.gsub(' ', '')}".constantize
+      form = form_class.new(parsed_form_data)
+
+      # This path can come about if the user is authenticated and, for some reason, doesn't have a participant_id
+      form.populate_veteran_data(@current_user) if form_id == 'vba_21_0966' && preparer_is_veteran?
+      form
+    end
+
+    def generate_filled_form
+      filler = SimpleFormsApi::PdfFiller.new(form_number: form_id, form:)
+      @current_user ? filler.generate(@current_user.loa[:current]) : filler.generate
+    end
+
+    def validate_metadata
+      SimpleFormsApiSubmission::MetadataValidator.validate(
+        form.metadata,
+        zip_code_is_us_based: form.zip_code_is_us_based
+      )
+    end
+
+    def preparer_is_veteran?
+      params[:preparer_identification] == 'VETERAN' && @current_user
+    end
+
+    def handle_form_specific_logic
+      if form_id == 'vba_20_10207'
+        @attachments = form.get_attachments
+      elsif %w[vba_40_0247 vba_40_10007].include?(form_id)
+        form.handle_attachments(file_path)
+      end
+    end
+
+    def upload_pdf
+      location, uuid = prepare_for_upload
+      log_upload_details(location, uuid)
+      response = use_benefits_intake_service? ? perform_pdf_upload(location) : perform_document_upload(location)
```

Comment:
I think we actually want to use [`SimpleFormsApi::PdfUploader`](https://github.com/department-of-veterans-affairs/vets-api/blob/b760b06482a687675d24e84e6797a1d47a661d16/modules/simple_forms_api/app/controllers/simple_forms_api/v1/uploads_controller.rb#L129) here, if the flipper is disabled.